File: app/src/main/java/com/example/worldnewsin/utils/YoutubeDataAPIService.java
package com.example.worldnewsin.utils;

import com.example.worldnewsin.SimpleVideoList;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface YoutubeDataAPIService {
    @GET("search")
    Call<SimpleVideoList> getSimpleVideosList(
            @Query("key") String key,
            @Query("channelId") String channelId,
            @Query("fields") String fields,
            @Query("part") String part,
            @Query("order") String order,
            @Query("maxResults") String maxResults
     );
}

File: app/src/main/java/online/litterae/worldnewsin/utils/YoutubeDataAPIService.kt
package online.litterae.worldnewsin.utils

import io.reactivex.rxjava3.core.Single
import online.litterae.worldnewsin.model.SimpleVideoList
import retrofit2.http.GET
import retrofit2.http.Query

interface YoutubeDataAPIService {
    @GET("search")
    fun getSimpleVideosList(
        @Query("key") key: String?,
        @Query("channelId") channelId: String?,
        @Query("fields") fields: String?,
        @Query("part") part: String?,
        @Query("order") order: String?,
        @Query("maxResults") maxResults: String?
    ): Single<SimpleVideoList>
}